pkg/manifests/manifests_test.go

```diff
@@ -1054,6 +1090,57 @@ ingress:
 	}
 }
 
+func TestAdditionalAlertManagerConfigs(t *testing.T) {
+	c, err := NewConfigFromString(`prometheusK8s:
+  additionalAlertManagerConfigs:
+  - apiVersion: v2
+    bearerToken:
+      name: alertmanager-bearer-token
+      key: token
+    scheme: https
+    tlsConfig:
+      ca: 
+        name: ca
+        key: root-ca.crt
+    pathPrefix: /
+    staticConfigs:
+    - alertmanager-remote.com
+    - alertmanager-remotex.com
+  - apiVersion: v2
+    bearerToken:
+      name: alertmanager-bearer-token
+      key: token
+    scheme: https
+    timeout: 60s
+    tlsConfig:
+      ca: 
+        name: ca
+        key: root-ca.crt
+      cert:
+        name: cert
+        key: cert.crt
+    pathPrefix: /
+    staticConfigs:
+    - alertmanager-remote.com
+`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c, defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath))
+
+	s, err := f.AdditionalAlertManagerConfigs()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	d := []AdditionalAlertmanagerConfig{}
+	err = yaml.Unmarshal(s.Data[AdditionalAlertmanagerConfigSecretKey], &d)
```

Comment:
hmm not sure why you deserialiaze the secret's data into `[]AdditionalAlertmanagerConfig`? I'd be ok with verifying that the resulting byte slice is the expected one.

Reply:
It is designed to deserialiaze the secret's data into `[]AlertmanagerConfig`. agreed with just verify that the resulting byte slice is the expected one. Updated.